### PR TITLE
livepatch: call apt-get update if snapd needs to be installed

### PR DESCRIPTION
**Pull request by blackboxsw**
Fixes: #691

### CODE REVIEW

**Comment by OddBloke** [comment]
@powersj We don't have an entry in the UX doc for this case (AFAICT), can you determine what the messaging we want to display in this case is?

File: uaclient/entitlements/livepatch.py [hunk 1]
```diff
@@ -50,6 +50,18 @@ def enable(self, *, silent_if_inapplicable: bool = False) -> bool:
...
+                    apt.run_apt_command(
+                        ["apt-get", "update"], status.MESSAGE_APT_UPDATE_FAILED
+                    )
+                except exceptions.UserFacingError as e:
+                    logging.debug(
+                        "Trying to install snapd."
+                        " Ignoring apt-get update failure: %s",
+                        str(e),
+                    )
+                    pass
```
Comment (OddBloke) [inline uaclient/entitlements/livepatch.py]: No need for this.

**Comment by powersj** [comment]
Hmm, from the UX doc here is the simple enable livepatch scenario:

```
$ sudo ua enable livepatch
One moment, checking your subscription first
Installing canonical-livepatch snap
A reboot is required to complete the install
```

However, in cases where we update the package lists we include the message, `Updating package lists`. Therefore, I propose:

```
$ sudo ua enable livepatch
One moment, checking your subscription first
Updating package lists
Installing canonical-livepatch snap
A reboot is required to complete the install
```

Or is there a concern we should be calling out the fact that we are installing systemd and snapd packages as well?

**Comment by OddBloke** [comment]
We already do emit "Installing snapd". That's what lead me to suspect that we maybe hadn't considered this case properly.

I'm not sure my opinion of what we should do here matters all that much, because I don't think we should be installing systemd on people's trusty systems full stop. So the best impression I can do of a user is "yes, please tell me that you're installing an entirely different init system so I can Ctrl-C out of this as fast as possible and reconsider the choices that brought me to this point", which presumably isn't the user experience we're optimising for. (And I do think people will (rightly) be *angry* if we install systemd on a trusty system without telling them. So... :man_shrugging:)

We did discuss at some point prompting for these installations, so perhaps we should emit the messages and they'll turn into prompts at some point in the future?

**Comment by powersj** [comment]
The existing behavior is to do the install without warning so I do not feel we are regressing or changing behavior for the worse. I do agree that adding an entirely new init system is a large change, which is why you successfully negotiated that we would not do this automatically. However, I say we merge this without the confirmation for now and we can loop back.

Please file a bug.

**Comment by OddBloke** [comment]
Yep, that sounds totally reasonable. I'll file two issues: one to cover the fact we don't mention that `systemd` is getting installed (with or without this PR), and one to cover prompting the user for permission.

**Comment by OddBloke** [comment]
OK, so @blackboxsw I think this just need that `pass` dropping and a rebase and we're good to go.

File: uaclient/entitlements/livepatch.py (hunk 1 above)
Comment (blackboxsw) [reply uaclient/entitlements/livepatch.py]: done.

**Pull request merged**